icefoganalytics/Travel-Authorization
# Fix Authorization Pipeline

# Context

Fix various problems (mostly around login):
- Login did not work at all
- login triggered 5-ish times each request
- non-logged in users where not booted out of the UI
- package-lock.json files where not commited to the repo

# Testing Instructions

1. Go to an inner url such as http://localhost:8080/my-travel-requests while unauthenticate.
2. You will be redirected to http://localhost:8080/sign-in.
3. Login and you will be redirected to http://localhost:8080/my-travel-requests.
4. This effect should be consistent across the app, and not through _any_ console errors.
5. Go to http://localhost:8080/sign-in?targetUrl=%2F.
6. This will not break, and instead redirect you to the dashboard.
7. Use "url navigation" by pasting in a new link e.g. http://localhost:8080/managerView
8. This will boot you to the sign-in page, because we aren't storing, and retrieving, auth cookies from localStorage yet.
